### DISCUSSION TITLE
Tangling to a different directory than the current working directory

### DISCUSSION THREAD

**Discussion by mjarus008** [nvim-neorg/neorg#1249] [Q&A]
Hi, I've been experimenting with the tangle feature and I can't seem to figure out how to tangle a specific code block to a path that I specify

Here's what I'm trying to do:

```norg
  #tangle code code/hello.go
  @code go
  package main

  import "fmt"

  func main() {
    fmt.Println("Hello, world.")
  }
  @end
```

But this produces the following error:
```
   Error  16:49:20 msg_show.lua_error   Neorg tangle current-file Error executing luv callback:
...nvim/lazy/neorg/lua/neorg/modules/core/tangle/module.lua:439: Failed to open file 'code / go.h' for tangling: ENOENT: no such file or directory: code / go.h
stack traceback:
	[C]: in function 'assert'
	...nvim/lazy/neorg/lua/neorg/modules/core/tangle/module.lua:439: in function <...nvim/lazy/neorg/lua/neorg/modules/core/tangle/module.lua:437>
```

**Comment by refaelsh** [comment]
I think it should be `#tangle code/hello.go` and not `#tangle code code/hello.go`.

**Reply by mjarus008** [reply]
@refaelsh nice catch, I didn't mean to post that version, however, I tried quite a few permutations and none seemed to work if `/` was present.
 so, `#tangle code/hello.go` on my box produces the following error:
```
   Error  09:50:43 msg_show.lua_error   Neorg tangle current-file Error executing luv callback:
...nvim/lazy/neorg/lua/neorg/modules/core/tangle/module.lua:438: Failed to open file '/hello.go' for tangling: EROFS: read-only file system: /hello.go
stack traceback:
	[C]: in function 'assert'
	...nvim/lazy/neorg/lua/neorg/modules/core/tangle/module.lua:438: in function <...nvim/lazy/neorg/lua/neorg/modules/core/tangle/module.lua:436>
```
What I noticed is that the presence of `/` changes the color of the the parameter passed to the `#tangle` Carryover tag, which makes me wonder if this is problem with the parser?

![image](https://github.com/nvim-neorg/neorg/assets/107445238/f899edc0-18ff-4d73-a4a6-9e1d35f2d4f9)


According to the spec, all characters (except line endings) can be used as parameters, [ref](https://github.com/nvim-neorg/norg-specs/blob/main/1.0-specification.norg#L796). So, I was expecting `code/hello.go` to be a valid parameter...

BTW even when I escape the `/` this still didn't work, i.e. if I did `#tangle code\/hello.go`

**Reply by refaelsh** [reply]
I am seeing the same issue as you describe while doing this:
```
#tangle ~/.config.nvim/init.lua
@code lua
bla bla bla bla
@end
```

Might it be that we both stumbled upon a bug? A penny for your thoughts :-)

**Reply by pysan3** [reply]
Could you check it with `tangle: code/hello.go` inside `@document.meta`?

This way of specifying should work, and if it does, it is probably a problem of the treesitter parser. We are currently developing a new version that should fix all the problems, so please use the document.meta method for now unfortunately. 

If specifying in meta also does not work, plz report it as it probably is a bug in tangle module.

**Reply by mjarus008** [reply]
I just tried out using the document metadata and that works. i.e. it produces the file in the right directory with the expected contents.

Unfortunately, for my current use case, I don't want to tangle all code blocks of the same file type to the same file so I've "hacked" the `tangle` module to treat white spaces as `/`, so I can do something like:

```norg
#tangle code go hello.go
@code go
...
@end
```

I had to add the following line to the `tangle` module ([this function](https://github.com/nvim-neorg/neorg/blob/main/lua/neorg/modules/core/tangle/module.lua#L419)) though:

```lua
-- replace white spaces with `/` for now
file = file:gsub("%s+", "/")

```

**Reply by pysan3** [reply]
Go ahead and join the development of the v3 parser ;)

**Reply by mjarus008** [reply]
Haha Im not an expert on these, just have an interest. I’m still learning but if an opportunity to contribute presents itself then I’d be happy to :)

**Comment by refaelsh** [comment]
@mjarus008 
> I don't want to tangle all code blocks of the same file type to the same file

I have the same problem :-)

**Reply by mjarus008** [reply]
Try the hack that I did for now until the parser is fixed?

**Reply by refaelsh** [reply]
No, this kind of ability is beyond me. I mainly do C++ and Haskell. I know nothing about Lua and/or Neovim plugin development.
